libpod/config/config.go

```diff
@@ -1,618 +0,0 @@
-		return errors.Wrap(err, _conmonVersionFormatErr)
-	}
-	major, err := strconv.Atoi(matches[1])
-	if err != nil {
-		return errors.Wrap(err, _conmonVersionFormatErr)
-	}
-	if major < _conmonMinMajorVersion {
-		return define.ErrConmonOutdated
-	}
-	if major > _conmonMinMajorVersion {
-		return nil
-	}
-
-	minor, err := strconv.Atoi(matches[2])
-	if err != nil {
-		return errors.Wrap(err, _conmonVersionFormatErr)
-	}
-	if minor < _conmonMinMinorVersion {
-		return define.ErrConmonOutdated
-	}
-	if minor > _conmonMinMinorVersion {
-		return nil
-	}
-
-	patch, err := strconv.Atoi(matches[3])
-	if err != nil {
-		return errors.Wrap(err, _conmonVersionFormatErr)
-	}
-	if patch < _conmonMinPatchVersion {
-		return define.ErrConmonOutdated
-	}
-	if patch > _conmonMinPatchVersion {
-		return nil
-	}
-
-	return nil
-}
-
-// NewConfig creates a new Config. It starts with an empty config and, if
-// specified, merges the config at `userConfigPath` path.  Depending if we're
-// running as root or rootless, we then merge the system configuration followed
-// by merging the default config (hard-coded default in memory).
-//
-// Note that the OCI runtime is hard-set to `crun` if we're running on a system
-// with cgroupsv2. Other OCI runtimes are not yet supporting cgroupsv2. This
-// might change in the future.
-func NewConfig(userConfigPath string) (*Config, error) {
-	// Start with the default config and iteratively merge fields in the system
-	// configs.
-	config, err := defaultConfigFromMemory()
-	if err != nil {
-		return nil, err
-	}
-
-	// Now, check if the user can access system configs and merge them if needed.
-	configs, err := systemConfigs()
-	if err != nil {
-		return nil, errors.Wrapf(err, "error finding config on system")
-	}
-
-	for _, path := range configs {
-		config, err = readConfigFromFile(path, config)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error reading system config %q", path)
-		}
-	}
-
-	// First, try to read the user-specified config
-	if userConfigPath != "" {
-		var err error
-		config, err = readConfigFromFile(userConfigPath, config)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error reading user config %q", userConfigPath)
-		}
-	}
-
-	// Since runc does not currently support cgroupV2
-	// Change to default crun on first running of libpod.conf
-	// TODO Once runc has support for cgroups, this function should be removed.
-	if !config.CgroupCheck && rootless.IsRootless() {
-		cgroupsV2, err := cgroups.IsCgroup2UnifiedMode()
-		if err != nil {
-			return nil, err
-		}
-		if cgroupsV2 {
-			path, err := exec.LookPath("crun")
-			if err != nil {
-				// Can't find crun path so do nothing
-				logrus.Warnf("Can not find crun package on the host, containers might fail to run on cgroup V2 systems without crun: %q", err)
-			} else {
-				config.CgroupCheck = true
-				config.OCIRuntime = path
-			}
-		}
-	}
-
-	// If we need to, switch to cgroupfs and logger=file on rootless.
-	config.checkCgroupsAndLogger()
```

Comment:
should use `func (c *Config) CheckCgroupsAndAdjustConfig()` directly in podman. https://github.com/containers/common/pull/80 removed it from `NewConfig`

Reply:
Good Catch,
Fixed